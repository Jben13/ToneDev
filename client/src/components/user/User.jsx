import React from 'react';
import axios from 'axios';
import Summary from '../SummaryView';

class User extends React.Component {
  constructor() {
    super();
    this.state = {
      SOUsername: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.submitSOname = this.submitSOname.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  submitSOname() {
    const { SOUsername } = this.state;
    axios.get('/api/user/so', {
      params: {
        username: SOUsername,
      },
    })
      .then((res) => {
        console.log(res.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }

  render() {
    return (
      <div>
        <form>
          <label htmlFor="SOuserName">
            StackOverflow Username:
            <input type="text" name="SOUsername" onChange={this.handleChange} />
          </label>
          <input type="button" value="Submit" onClick={this.submitSOname} />
        </form>
        <Summary />

      </div>
    );
  }
}

export default User;
