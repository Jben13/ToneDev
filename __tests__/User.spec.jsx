import React from 'react';
import User from '../client/src/components/user/User';

test('App renders component correctly', () => {
  const component = global.shallow(<User />);
  expect(component).toMatchSnapshot();
});
