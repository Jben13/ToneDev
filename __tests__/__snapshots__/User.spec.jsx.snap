// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  <form>
    <label
      htmlFor="SOuserName"
    >
      Name:
      <input
        name="name"
        type="text"
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>
  <Summary />
</div>
`;
